File: widgetssdk/src/main/java/com/glia/widgets/core/survey/GliaSurveyRepository.java
package com.glia.widgets.core.survey;

import androidx.annotation.NonNull;

import com.glia.androidsdk.GliaException;
import com.glia.androidsdk.engagement.Survey;
import com.glia.widgets.di.GliaCore;

import java.util.List;
import java.util.function.Consumer;

public class GliaSurveyRepository {

    private final GliaCore gliaCore;

    public GliaSurveyRepository(GliaCore gliaCore) {
        this.gliaCore = gliaCore;
    }

    public void submitSurveyAnswers(@NonNull List<Survey.Answer> answers,
                                    @NonNull String surveyId,
                                    @NonNull String engagementId,
                                    @NonNull Consumer<GliaException> callback) {
        gliaCore.submitSurveyAnswers(answers, surveyId, engagementId, callback);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/internal/survey/GliaSurveyRepository.kt
package com.glia.widgets.internal.survey

import com.glia.androidsdk.GliaException
import com.glia.androidsdk.engagement.Survey
import com.glia.widgets.di.GliaCore
import com.glia.widgets.helper.Logger
import com.glia.widgets.helper.TAG
import java.util.function.Consumer

internal class GliaSurveyRepository(private val gliaCore: GliaCore) {
    fun submitSurveyAnswers(
        answers: List<Survey.Answer>,
        surveyId: String,
        engagementId: String,
        callback: Consumer<GliaException?>
    ) {
        Logger.i(TAG, "Submit survey answers")
        gliaCore.submitSurveyAnswers(answers, surveyId, engagementId, callback)
    }
}
